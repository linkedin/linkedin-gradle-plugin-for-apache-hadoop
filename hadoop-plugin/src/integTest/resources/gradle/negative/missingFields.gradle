plugins {
  id 'com.linkedin.gradle.hadoop.HadoopPlugin'
}

// Negative test for missing required fields.
hadoop {
  buildPath "jobs"
  cleanPath false

  workflow('missingFieldsWorkflow1') {
    propertyFile('properties1') {
    }

    job('job1') {
    }

    commandJob('job2') {
      depends 'job1'
    }

    hadoopJavaJob('job3') {
      depends 'job2'
    }

    hiveJob('job4') {
      depends 'job3'
    }

    javaJob('job5') {
      depends 'job5'
    }

    javaProcessJob('job6') {
      depends 'job5'
    }

    kafkaPushJob('job7') {
      depends 'job6'
    }

    noOpJob('job8') {
      depends 'job7'
    }

    pigJob('job9') {
      depends 'job8'
    }

    sparkJob('job10') {
      depends 'job9'
    }

    sparkJob('job10Jar1') {
      executes 'hello-spark.jar'
      depends 'job10'
    }

    sparkJob('job10Jar2') {
      uses ''
      executes 'hello-spark.jar'
      depends 'job10Jar1'
    }

    sparkJob('job10Py') {
      uses 'com.linkedin.HelloSpark'
      executes 'hello_pyspark.py'
      depends 'job10Jar2'
    }

    voldemortBuildPushJob('job11') {
      depends 'job10Py'
    }

    hadoopShellJob('job12') {
      depends 'job11'
    }

    hdfsToTeradataJob('job13') {
      depends 'job12'
    }

    teradataToHdfsJob('job14') {
      depends 'job13'
    }

    hdfsToEspressoJob('job15') {
      depends 'job14'
    }

    gobblinJob('job16') {
      depends 'job15'
    }

    sqlJob('job17') {
      depends 'job16'
    }

    pinotBuildAndPushJob('job18') {
      depends 'job17'
    }

    tableauJob('job19') {
      depends 'job18'
    }

    hdfsWaitJob('job20') {
      depends 'job19'
    }

    venicePushJob('job23') {
      depends 'job20'
    }

    tensorFlowJob('job24', 'tony') {
      depends 'job23'
    }

    tensorFlowJob('job25', 'spark') {
      depends 'job24'
    }

    tonyJob('job25a') {
      depends 'job25'
    }

    wormholePushJob('job26') {
      depends 'job25a'
    }

    kabootarJob('job27') {
      depends 'job26'
    }

    wormholePushJob2('job28') {
      depends 'job27'
    }

    kubernetesJob('job29') {
      depends 'job28'
    }

    carbonJob('job30') {
        depends 'job29'
    }

    targets 'job30'
  }
}
