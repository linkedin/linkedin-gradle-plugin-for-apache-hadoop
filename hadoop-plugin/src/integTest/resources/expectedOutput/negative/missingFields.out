JavaJob has been deprecated in favor of HadoopJavaJob or JavaProcessJob. Please change the job job5 to one of these classes.
RequiredFieldsChecker ERROR: CommandJob job2 must set command or commands
RequiredFieldsChecker ERROR: HadoopJavaJob job3 must set jobClass
RequiredFieldsChecker ERROR: HiveJob job4 must set script
RequiredFieldsChecker ERROR: JavaJob job5 must set jobClass
RequiredFieldsChecker ERROR: JavaProcessJob job6 must set javaClass
RequiredFieldsChecker ERROR: KafkaPushJob job7 must set inputPath
RequiredFieldsChecker ERROR: KafkaPushJob job7 must set topic
RequiredFieldsChecker ERROR: PigJob job9 must set script
RequiredFieldsChecker ERROR: SparkJob job10 must set executes
RequiredFieldsChecker ERROR: SparkJob job10Jar1 must set uses for Java application
RequiredFieldsChecker ERROR: SparkJob job10Jar2 must set uses for Java application
RequiredFieldsChecker ERROR: SparkJob job10Py must not set uses for Python application
RequiredFieldsChecker ERROR: VoldemortBuildPushJob job11 has the following required fields: storeName, clusterName, buildInputPath, buildOutputPath, storeOwners, storeDesc
RequiredFieldsChecker ERROR: HadoopShellJob job12 must set command or commands
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 has the following required fields: hostName
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 has the following required fields: userId
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 has the following required fields: targetTable
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 needs either credentialName or encryptedCredential defined
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 needs either avroSchemaPath or avroSchemaInline defined
RequiredFieldsChecker ERROR: HdfsToTeradataJob job13 has the following required fields: sourceHdfsPath
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 has the following required fields: hostName
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 has the following required fields: userId
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 has the following required fields: targetHdfsPath
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 needs either credentialName or encryptedCredential defined
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 needs either sourceTable or sourceQuery defined
RequiredFieldsChecker ERROR: TeradataToHdfsJob job14 needs either avroSchemaPath or avroSchemaInline defined
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following error: qps should be a positive number
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: sourceHdfsPath
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoEndpoint
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoDatabaseName
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoTableName
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoContentType
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoKey
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: espressoOperation
RequiredFieldsChecker ERROR: HdfsToEspressoJob job15 has the following required fields: errorHdfsPath
RequiredFieldsChecker ERROR: SqlJob job17 has the following required fields: jdbcUrl
RequiredFieldsChecker ERROR: SqlJob job17 has the following required fields: jdbcUserId
RequiredFieldsChecker ERROR: SqlJob job17 has the following required fields: jdbcEncryptedCredential
RequiredFieldsChecker ERROR: SqlJob job17 has the following required fields: jdbcCryptoKeyPath
RequiredFieldsChecker ERROR: PinotBuildAndPushJob job18 must set tableName, inputPath, and pushLocation
RequiredFieldsChecker ERROR: TableauJob job19 must set workbookName
RequiredFieldsChecker ERROR: HdfsWaitJob job20 must set dirPath, freshness, timeout, and forceJobToFail
RequiredFieldsChecker ERROR: VenicePushJob job23 must set avroKeyField, avroValueField, clusterName, inputPath, veniceStoreName
RequiredFieldsChecker ERROR: WormholePushJob job26 must set inputPath, namespace and dataset
RequiredFieldsChecker ERROR: KabootarJob job27 must set trainingModelLocation, trainingName, aiProjectGroup, wormholeNamespace, initialImport. Please see the job documentation for more details.
RequiredFieldsChecker ERROR: WormholePushJob2 job28 must set inputPath, namespace and dataset
RequiredFieldsChecker ERROR: KubernetesJob job29 has the following required fields: kind
RequiredFieldsChecker ERROR: CarbonJob job30 must set task.type
RequiredFieldsChecker WARNING: Properties properties1 does not set any confProperties, jobProperties, jvmProperties or basePropertySetName. Nothing will be built for this properties object.
